Assets/_Scripts/BattleFight.cs

```diff
@@ -77,16 +77,16 @@ public class BattleFight : IDisposable
             Debug.LogError("Cannot find queued action: "  + attackerName);
             yield break;
         }
-        string attackName = attackerAction._action.name;
-        List<FightingEntity> targets = attackerAction._action.GetTargets(this.fighter, attackerAction.GetTargetIds());
-
-        yield return GameManager.Instance.time.StartCoroutine(doAnimation(this.fighter, attackerAction, targets));
-
-        FightResult fightResult = attackerAction.ExecuteAction();
-
-        this.CheckIfAnyEntityDied(fightResult, attackerAction._action);
-
+        ActionBase action = attackerAction.GetAction();
+        string attackName = action.name;
+        List<FightingEntity> targets = action.GetTargets(this.fighter, attackerAction.GetTargetIds());
+        yield return GameManager.Instance.time.GetController().StartCoroutine(action.ExecuteAction(fighter, targets));
+        FightResult fightResult = action.lastFightResult;
+        this.CheckIfAnyEntityDied(fightResult, action);
         this.onFightEnd(fightResult);
+        if (action.animation != null) {
+            yield return GameManager.Instance.time.GetController().WaitForSeconds(action.animation.timeAfterEffect);
```

Comment:
is it possible for us to wait for `timeAfterEffect` in the same function (`ExecuteAction`) as `timeBeforeEffect`? Would look a bit cleaner.